Kconfig.zephyr

```diff
@@ -248,6 +248,27 @@ config COMPILER_OPT
 
 endmenu
 
+choice
+	prompt "Error checking behavior"
+	default RUNTIME_ERROR_CHECKS
+
+config ASSERT_ON_ERRORS
+	bool "Assert on all errors"
+	help
+	  Assert on errors covered with the CHECK macro.
+
+config NO_RUNTIME_CHECKS
+	bool "No runtime error checks"
+	help
+	  Do not do any runtime checks or asserts.
+
+config RUNTIME_ERROR_CHECKS
+	bool "Enable runtime error checks"
+	help
+	  Do runtime checks, this is the default and the only option used
```

Comment:
Nit: Maybe better rephrase "and the only option used" => "and the typical option used"
I guess we will need sometimes to disable input checks for some reasons (e.g. sample almost too big for microbit)

Reply:
This is going to be the only option enabled during testing, there is no way around it. Enable any of the other options and things will collapse now that we are going to do negative testing, the whole purpose of this action.

Reply:
If this is indeed the only option used during testing, things will probably be broken quite soon for the other options. I would make sense to do most of the kernel testing with this one. But still, I would consider it reasonable to have a couple of tests, if only tests with of samples or higher functionality, which would use the other options.